### PR TITLE
MM-47294 : Migrate /custom_status_expiry e2e tests to TypeScript

### PR DESCRIPTION
**Pull request by satya-vinay**
#### Summary

Converted the following spec files under /integration/custom_status/custom_status_expiry to Typescript

1. custom_status_expiry_1_spec.js
2. custom_status_expiry_2_spec.js
3. custom_status_expiry_3_spec.js
4. custom_status_expiry_4_spec.js

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/21314
JIRA: https://mattermost.atlassian.net/browse/MM-47294

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by AshishDhama** [comment]
@satya-vinay could you please check if
> 2. Move over all the types from the declaration (d.ts) files into the matching typescript file, and delete the declaration file (see Support Migration PR link below for reference )

Is possible here, I can see you have added a new interface in the `index.d.ts`? As we are planning to remove `d.ts` (type declaration files) I don’t think this will be ideal.

**Comment by satya-vinay** [comment]
> @satya-vinay could you please check if
> 
> > 2. Move over all the types from the declaration (d.ts) files into the matching typescript file, and delete the declaration file (see Support Migration PR link below for reference )
> 
> Is possible here, I can see you have added a new interface in the `index.d.ts`? As we are planning to remove `d.ts` (type declaration files) I don’t think this will be ideal.

@AshishDhama Sorry, my bad. I have added the new changes. Please check.

**Comment by AshishDhama** [comment]
@furqanmlk If you take a look. Thanks

File: e2e/cypress/tests/integration/custom_status/custom_status_expiry/custom_status_expiry_1_spec.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 // Stage: @prod
 // Group: @custom_status
+import dayjs from 'dayjs';
```
Comment (furqanmlk) [inline e2e/cypress/tests/integration/custom_status/custom_status_expiry/custom_status_expiry_1_spec.ts:12]:
```suggestion

import dayjs from 'dayjs';
```

File: e2e/cypress/tests/integration/custom_status/custom_status_expiry/custom_status_expiry_2_spec.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@
 
 // Stage: @prod
 // Group: @custom_status
+import dayjs from 'dayjs';
```
Comment (furqanmlk) [inline e2e/cypress/tests/integration/custom_status/custom_status_expiry/custom_status_expiry_2_spec.ts:12]:
```suggestion

import dayjs from 'dayjs';
```

File: e2e/cypress/tests/integration/custom_status/custom_status_expiry/custom_status_expiry_3_spec.ts [hunk 3]
```diff
@@ -9,6 +9,7 @@
 
 // Stage: @prod
 // Group: @custom_status
+import dayjs from 'dayjs';
```
Comment (furqanmlk) [inline e2e/cypress/tests/integration/custom_status/custom_status_expiry/custom_status_expiry_3_spec.ts:12]:
```suggestion

import dayjs from 'dayjs';
```

File: e2e/cypress/tests/integration/custom_status/custom_status_expiry/custom_status_expiry_4_spec.ts [hunk 4]
```diff
@@ -9,6 +9,7 @@
 
 // Stage: @prod
 // Group: @custom_status
+import dayjs from 'dayjs';
```
Comment (furqanmlk) [inline e2e/cypress/tests/integration/custom_status/custom_status_expiry/custom_status_expiry_4_spec.ts:12]:
```suggestion

import dayjs from 'dayjs';
```

**Review by furqanmlk — APPROVED** [review]
Thanks @satya-vinay 
LGTM !
Please update with minor changes

**Comment by satya-vinay** [comment]
@furqanmlk Thanks for the feedback. I have added the requested minor changes, please have a look

**Pull request merged**